src/lib/structures/music/Queue.ts

```diff
@@ -135,6 +136,18 @@ export class Queue extends Array<Song> {
 		return this;
 	}
 
+	public toJSON() {
+		return {
+			length: this.length,
+			voiceChannel: this.voiceChannel,
+			playing: this.playing,
+			song: this.song,
+			position: this.position,
+			status: this.status,
+			queue: [...this]
```

Comment:
Won't this cause circulars unless you call toJSON on each song?

Reply:
Nope.